docs/caching.md

```diff
@@ -78,27 +46,94 @@ const myRouter = [
 
-We should avoid needing to hand-manage individual Redis cache entries, but occasionally issues arise where an AE or partner posts a bug saying that they can't see their new feature that they've created, or a change they've made somewhere to an artwork (while logged out). If this happens:
+Since logged-in queries always propagate the user's access token, they will _never_ be cached by the CDN, unless they are opted-in.
 
-- Visit https://artsy.net/admin/cache
-- Click 'load all cache keys'
-- Enter the id (or some other piece of metadata) for the page that needs an update. (eg, artist/artwork slug, or some other id.)
-- Click 'Delete cache key'
-- Tell AE to reload page, which will return the latest version
+For logged-in queries, there is an opt-in directive `@cacheable` that can be added to a query. When present - no access token will be propagated with the request, and the `no-cache` value in the `Cache-Control` header will not be included - which will thus allow the CDN to cache this query. This is useful for queries that are not user-specific, and can be cached for all users.
+
+Worth noting that one can mix and match, and add `@cacheable` _and_ a `force: true` configuration to the same query (there is no validation preventing this). As a conservative choice, when there is a conflict between an _opt-in_ (via `@cacheable`) and an _opt-out_ (via `force: true`), the opt-out takes precedence (and thus this query would not be cached).
+
+**What if I want to see the latest page / unpublished preview?** There is an escape hatch - a query param in the URL: `?nocache=true` - that will force the CDN to bypass the cache and hit Metaphysics directly _despite_ any `@cacheable` directive. This can be useful for a privileged user to see an unpublished page (which we wouldn't want to cache).
+
+Setting/unsetting the `Cache-Control` header, and propagating any `max-age` is accomplished in a Relay middleware: [`System/Relay/middleware/cacheHeaderMiddleware.tsx`](System/Relay/middleware/cacheHeaderMiddleware.tsx). This takes into account the various configuration options available (including `@cacheable`).
+
+Un-setting the access token header for logged-in queries using `@cacheable` is accomplished in a Relay middleware: [`System/Relay/createRelaySSREnvironment.ts`](System/Relay/createRelaySSREnvironment.ts).
+
+### Cache-related ENV vars for Relay caching:
+
+- `GRAPHQL_CACHE_SIZE`: Total number of entries before eviction (on the client)
+- `GRAPHQL_CACHE_TTL`: Default expiration TTL in milliseconds (on the client) - defaults when unset to 1 hour
+
+### Relay Store Normalization Issues with Prefetching and CDN Cache
+
+Sometimes issues arise where a prefetched and cached network response conflicts with fresh uncached data (via, say, a QueryRenderer). In rare cases like this Relay often fails to normalize the data, which can create conflicts in the store and possible runtime errors.
```

Comment:
**thought (non-blocking):** Slightly confusing whether this should be regarded as "rare" or something that happens "often" (or maybe this is correct as-is because it is a likely consequence of an unlikely event 😵‍💫)